### PR TITLE
[develop]: Updated ConfigWorkflow.rst to reflect changes to config_defaults.yaml (PI13)

### LINKED ISSUE #1132: [develop]: Updated ConfigWorkflow.rst to reflect changes to config_defaults.yaml (PI13)
Periodically, documentation needs to be updated to reflect the most recent changes to the repository. Generally, updates include:

Edits to ConfigWorkflow.rst to align it with config_defaults.yaml

Solution
Update documentation to reflect any recent changes.

### PR DESCRIPTION
**Pull request by jdkublnick**
DESCRIPTION OF CHANGES:
Updated ConfigWorkflow.rst to reflect recent changes to config_defaults.yaml in order to keep documentation up to date.
Type of change
[X ] This change requires a documentation update
TESTS CONDUCTED:
None required for documentation. The docs build successfully on my fork and can be viewed at: https://ufs-srweather-app-joshua-kublnick.readthedocs.io/en/text-us-336/BackgroundInfo/index.html
DEPENDENCIES:
N/A

DOCUMENTATION:
All documentation

ISSUE:
Issue https://github.com/ufs-community/ufs-srweather-app/issues/1132

CHECKLIST
[X ] My code follows the style guidelines in the Contributor's Guide
[X ] I have performed a self-review of my own code using the Code Reviewer's Guide
 I have commented my code, particularly in hard-to-understand areas
[X ] My changes need updates to the documentation. I have made corresponding changes to the documentation
 My changes do not require updates to the documentation (explain).
[X ] My changes generate no new warnings
 New and existing tests pass with my changes
[ X] Any dependent changes have been merged and published
LABELS (optional):
A Code Manager needs to add the following labels to this PR:

[X ] documentation

### CODE REVIEW

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 1]
```diff
@@ -159,6 +159,9 @@ These settings define platform-specific run commands. Users should set run comma
 ``RUN_CMD_AQM``: (Default: "")
   The run command for some AQM tasks.
 
+``RUN_CMD_NEXUS``: (Default: "")
+  The run command for the AQM NEXUS tasks.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop.

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 2]
```diff
@@ -271,6 +274,12 @@ These parameters are associated with the fixed (i.e., static) files. On :srw-wik
 ``FIXshp``: (Default: "")
    System directory containing the graphics shapefiles. On Level 1 systems, these are set within the machine files. Users on other systems will need to provide the path to the directory that contains the *Natural Earth* shapefiles.
 
+``FIXaqm``: (Default: "")
+   System directory where AQM data files are located.
+
+``FIXemis``: (Default: "")
+   System directory where AQM emission data files are located.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: Both of these entries are already present in `ConfigWorkflow.rst` at the HEAD of develop

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 3]
```diff
@@ -389,6 +398,9 @@ Set File Name Parameters
 ``NEMS_CONFIG_FN``: (Default: "nems.configure")
    Name of a file that contains information about the various :term:`NEMS` components and their run sequence. In general, users should not set this variable in their configuration file (see :ref:`note <tmpl-fn-warning>`).
 
+``UFS_CONFIG_FN``: (Default: "ufs.configure")
+  Name of a template file that contains information about the various UFS components and their run sequence (ufs-weather-model: ufs.configure). Its default value is the name of the file that the ufs weather model expects to read in.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop.  This configuration replaces the `NEMS_CONFIG_FN` option above.

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 4]
```diff
@@ -422,6 +434,9 @@ Set File Path Parameters
 ``NEMS_CONFIG_TMPL_FP``: (Default: ``'{{ [user.PARMdir, NEMS_CONFIG_FN]|path_join }}'``) 
    Path to the ``NEMS_CONFIG_FN`` file. 
 
+``UFS_CONFIG_TMPL_FP:``: (Default: ``'{{ [user.PARMdir, UFS_CONFIG_FN]|path_join }}'``)
+   Path to the ``UFS_CONFIG_FN`` file.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop. This configuration replaces the `NEMS_CONFIG_TMPL_FP` option above.

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 5]
```diff
@@ -440,6 +455,9 @@ This section contains files and paths to files that are staged in the experiment
 ``NEMS_CONFIG_FP``: (Default: ``'{{ [EXPTDIR, NEMS_CONFIG_FN]|path_join }}'``)
    Path to the ``NEMS_CONFIG_FN`` file in the experiment directory. 
 
+``UFS_CONFIG_FP``: (Default: ``'{{ [EXPTDIR, UFS_CONFIG_FN]|path_join }}'``)
+   Path to the ``UFS_CONFIG_FN`` file in the experiment directory.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop. This configuration replaces the `NEMS_CONFIG_FP` option above.

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 6]
```diff
@@ -452,8 +470,8 @@ This section contains files and paths to files that are staged in the experiment
...
-   Name of the file (a shell script) containing definitions of the primary and secondary experiment variables (parameters). This file is sourced by many scripts (e.g., the J-job scripts corresponding to each workflow task) in order to make all the experiment variables available in those scripts. The primary variables are defined in the default configuration file (``config_defaults.yaml``) and in the user configuration file (``config.yaml``). The secondary experiment variables are generated by the experiment generation script. 
+``GLOBAL_VAR_DEFNS_FN``: (Default: "var_defns.yaml")
+   Name of the experiment configuration file. It contains the primary experiment variables defined in this default configuration script and in the user-specified configuration as well as secondary experiment variables generated by the experiment generation script. This file is the primary source of information used in the scripts at run time. The primary variables are defined in the default configuration file (``config_defaults.yaml``) and in the user configuration file (``config.yaml``). The secondary experiment variables are generated by the experiment generation script.
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 7]
```diff
@@ -532,13 +550,13 @@ CCPP Parameter
 ``CCPP_PHYS_SUITE_FP``: (Default: ``'{{ [workflow.EXPTDIR, CCPP_PHYS_SUITE_FN]|path_join }}'``)
    The full path to the suite definition file (SDF) in the experiment directory. 
 
-``CCPP_PHYS_DIR``: (Default: ``'{{ [user.UFS_WTHR_MDL_DIR, "FV3", "ccpp", "physics", "physics"] |path_join }}'``)
+``CCPP_PHYS_DIR``: (Default: ``'{{ [user.UFS_WTHR_MDL_DIR, "FV3", "ccpp", "physics", "physics", "SFC_models", "Land", "Noahmp"] |path_join }}'``)
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop

File: doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 8]
```diff
@@ -532,13 +550,13 @@ CCPP Parameter
...
    The directory containing the CCPP physics source code. This is needed to link table(s) contained in that repository. 
 
 Field Dictionary Parameters
 ------------------------------
 
-``FIELD_DICT_FN``: (Default: "fd_nems.yaml")
+``FIELD_DICT_FN``: (Default: "fd_ufs.yaml")
```
Comment (MichaelLueken) [inline doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is already present in `ConfigWorkflow.rst` at the HEAD of develop

**Review by MichaelLueken — CHANGES_REQUESTED** [review]
@jdkublnick -

Thanks for going through the `ConfigWorkflow.rst` documentation and updating the entries that have been updated in `config_defaults.yaml`!

Cross referencing the changes that you have made in this PR with what is currently in the `ConfigWorkflow.rst` file at the HEAD of the develop branch, it looks like several of these entries are currently in the documentation.  Looking at the commit history for your `text/us-336` branch, it looks like the branch was created from a February 21, 2024 hash, with no other syncing with the develop branch after it was created.

Please merge the authoritative develop branch back into your `text/us-336` branch.  If you have any questions or require assistance, please let me know.

@gspetro-NOAA - While looking through the `ConfigWorkflow.rst` file and `config_defaults.yaml`, specifically in the `FIX*` sections, there is a disconnect occurring.  In `ConfigWorkflow.rst`, the definition of variables is `path to system directory`, while in `config_defaults.yaml`, it is `system directory where`.

Should the definitions be updated in either the `ConfigWorkflow.rst` or `config_defaults.yaml` files (and which file should be updated, if you think it should), or do you feel that the differences for the `FIX*` variables are fine as is?

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 9]
```diff
@@ -159,6 +159,9 @@ These settings define platform-specific run commands. Users should set run comma
 ``RUN_CMD_NEXUS``: (Default: "")
   The run command for the AQM NEXUS tasks.
 
+``RUN_CMD_NEXUS``: (Default: "")
+  The run command for the AQM NEXUS tasks.
+
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]:
```suggestion
```

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 10]
```diff
@@ -272,10 +275,17 @@ These parameters are associated with the fixed (i.e., static) files. On :srw-wik
...
+   System directory where AQM data files are located.
+
+``FIXemis``: (Default: "")
+   System directory where AQM emission data files are located.
+=======
    Path to system directory containing AQM fixed files.
 
 ``FIXemis``: (Default: "")
    Path to system directory containing AQM emission data files.
+>>>>>>> origin/develop:doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: This is a merge conflict that needs to be resolved.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 11]
```diff
@@ -293,7 +303,11 @@ WORKFLOW Configuration Parameters
 
 If non-default parameters are selected for the variables in this section, they should be added to the ``workflow:`` section of the ``config.yaml`` file. 
 
+<<<<<<< HEAD:doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst
+``WORKFLOW_ID``: (Default: "")
+=======
 ``WORKFLOW_ID``: (Default: ``''``)
+>>>>>>> origin/develop:doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: Another merge conflict that needs to be resolved.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 12]
```diff
@@ -395,6 +409,9 @@ Set File Name Parameters
 ``UFS_CONFIG_FN``: (Default: "ufs.configure")
    Name of a file that contains information about the various :term:`UFS` components and their run sequence. In general, users should not set this variable in their configuration file (see :ref:`note <tmpl-fn-warning>`).
 
+``UFS_CONFIG_FN``: (Default: "ufs.configure")
+  Name of a template file that contains information about the various UFS components and their run sequence (ufs-weather-model: ufs.configure). Its default value is the name of the file that the ufs weather model expects to read in.
+
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]:
```suggestion
```
This variable is listed above.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 13]
```diff
@@ -428,6 +445,9 @@ Set File Path Parameters
 ``UFS_CONFIG_TMPL_FP``: (Default: ``'{{ [user.PARMdir, UFS_CONFIG_FN]|path_join }}'``) 
    Path to the ``UFS_CONFIG_FN`` file. 
 
+``UFS_CONFIG_TMPL_FP:``: (Default: ``'{{ [user.PARMdir, UFS_CONFIG_FN]|path_join }}'``)
+   Path to the ``UFS_CONFIG_FN`` file.
+
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]:
```suggestion
```
This variable is listed above.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 14]
```diff
@@ -446,6 +466,9 @@ This section contains files and paths to files that are staged in the experiment
 ``UFS_CONFIG_FP``: (Default: ``'{{ [EXPTDIR, UFS_CONFIG_FN]|path_join }}'``)
    Path to the ``UFS_CONFIG_FN`` file in the experiment directory. 
 
+``UFS_CONFIG_FP``: (Default: ``'{{ [EXPTDIR, UFS_CONFIG_FN]|path_join }}'``)
+   Path to the ``UFS_CONFIG_FN`` file in the experiment directory. 
+
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]:
```suggestion
```
Variable already included

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 15]
```diff
@@ -459,7 +482,11 @@ This section contains files and paths to files that are staged in the experiment
...
+<<<<<<< HEAD:doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst
+   Name of the experiment configuration file. It contains the primary experiment variables defined in this default configuration script and in the user-specified configuration as well as secondary experiment variables generated by the experiment generation script. This file is the primary source of information used in the scripts at run time. The primary variables are defined in the default configuration file (``config_defaults.yaml``) and in the user configuration file (``config.yaml``). The secondary experiment variables are generated by the experiment generation script. 
+=======
    Name of the auto-generated experiment configuration file. It contains the primary experiment variables defined in this default configuration script and in the user-specified configuration as well as secondary experiment variables generated by the experiment generation script from machine files and other settings. This file is the primary source of information used in the scripts at run time.
+>>>>>>> origin/develop:doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: Merge conflict that needs to be resolved.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 16]
```diff
@@ -538,7 +565,11 @@ CCPP Parameter
 ``CCPP_PHYS_SUITE_FP``: (Default: ``'{{ [workflow.EXPTDIR, CCPP_PHYS_SUITE_FN]|path_join }}'``)
    The full path to the suite definition file (SDF) in the experiment directory. 
 
+<<<<<<< HEAD:doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst
+``CCPP_PHYS_DIR``: (Default: ``'{{ [user.UFS_WTHR_MDL_DIR, "FV3", "ccpp", "physics", "physics", "SFC_models", "Land", "Noahmp"] |path_join }}'``)
+=======
 ``CCPP_PHYS_DIR``: (Default: ``'{{ [user.UFS_WTHR_MDL_DIR, "FV3", "ccpp", "physics", "physics", "SFC_Models", "Land", "Noahmp"] |path_join }}'``)
+>>>>>>> origin/develop:doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: Merge conflict that needs to be resolved.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 17]
```diff
@@ -1096,10 +1127,15 @@ For each workflow task, certain parameter values must be passed to the job sched
 
       For more information, see the `Intel Development Reference Guide <https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-10/thread-affinity-interface.html>`__. 
 
+<<<<<<< HEAD:doc/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst
+``OMP_NUM_THREADS_RUN_FCST``: (Default: 2)
+   The number of OpenMP threads to use for parallel regions. Corresponds to the ``ATM_omp_num_threads`` value in ``nems.configure``.
+=======
 ``OMP_NUM_THREADS_RUN_FCST``: (Default: 1)
    The number of OpenMP threads to use for parallel regions. Corresponds to the ``atmos_nthreads`` value in ``model_configure``.
+>>>>>>> origin/develop:doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: Merge conflict that needs to be resolved.

**Comment by gspetro-NOAA** [comment]
@jdkublnick I think the only changes required are from [PR #1098](https://github.com/ufs-community/ufs-srweather-app/commit/c377164582ee071ce8b3921e10b2d0f100141887#diff-a4384c6e3a35bddb4828a36142825b542f035fac5ea6df0e7b72e1c2a56bb4e5). Pretty much everything else should remain as-is.

**Comment by gspetro-NOAA** [comment]
@MichaelLueken I think "Path to system directory..." is more accurate than "System directory where...". As far as I can tell, users would need to add the full path to the directory, not just the name of the directory, and the "Path to..." makes that clearer imo. Looking at, e.g., the [Hera machine file](https://github.com/ufs-community/ufs-srweather-app/blob/develop/ush/machine/hera.yaml), `FIXaqm` and similar variables are defined using a full path. I presume a user would need to change the path, not just the directory name, if their data is located somewhere else on their system. In that respect, it could be better to update the `config_defaults.yaml` file comments, but I have hesitated to overwrite what developers have put in. That said, the RST files are more clearly within our purview, and I have improved definitions where possible.

**Comment by MichaelLueken** [comment]
Thanks for the explanation, @gspetro-NOAA!  I'm fine with keeping `config_defaults.yaml` as it currently is and providing more detailed information in the User Guide documentation.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 18]
```diff
@@ -1099,7 +1099,7 @@ For each workflow task, certain parameter values must be passed to the job sched
 ``OMP_NUM_THREADS_RUN_FCST``: (Default: 1)
```
Comment (MichaelLueken) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]: The defaults for OMP_NUM_THREADS_RUN_FCST should be 2.

```suggestion
``OMP_NUM_THREADS_RUN_FCST``: (Default: 2)
```

**Review by MichaelLueken — COMMENTED** [review]
@jdkublnick 

Thanks for merging the latest develop into text/us-336!  The `OMP_NUM_THREADS_RUN_FCST` default value should be 2, rather than 1.

**Review by MichaelLueken — APPROVED** [review]
Thanks again, @jdkublnick!

The differences between the `config_defaults.yaml` and `ConfigWorkflow.rst` files appear to be addressed now.

File: doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst [hunk 19]
```diff
@@ -393,7 +393,7 @@ Set File Name Parameters
    Name of a file that contains settings and configurations for the :term:`NUOPC`/:term:`ESMF` main component. In general, users should not set this variable in their configuration file (see :ref:`note <tmpl-fn-warning>`).
 
 ``UFS_CONFIG_FN``: (Default: "ufs.configure")
-   Name of a file that contains information about the various :term:`UFS` components and their run sequence. In general, users should not set this variable in their configuration file (see :ref:`note <tmpl-fn-warning>`).
+  Name of a template file that contains information about the various UFS components and their run sequence (ufs-weather-model: ufs.configure). Its default value is the name of the file that the ufs weather model expects to read in.
```
Comment (gspetro-NOAA) [inline doc/UsersGuide/CustomizingTheWorkflow/ConfigWorkflow.rst]:
```suggestion
   Name of a template file that contains information about the various :term:`UFS` components and their run sequence. In general, users should not set this variable in their configuration file (see :ref:`note <tmpl-fn-warning>`).
```
@jdkublnick I'd like to revert this definition to what is in `develop` already, tho you can keep the addition of the word "template." The default value is already listed, and the extra info in the `develop` RST file is important.

**Review by gspetro-NOAA — APPROVED** [review]
Requested changes were made. Looks good to me.

**Comment by MichaelLueken** [comment]
Moving forward with merging this work now.

**Pull request merged**